### DISCUSSION TITLE
Assets index page should have an action menu below its source list

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/cms#8503] [Ideas]
On the Assets  index page, when a folder is selected in the source list (as opposed to a plugin-defined source), an action menu should appear below the source list, with the same options you get when you right-click on the folder.

Visually it should look similar to the action menu in the sidebar on Settings → Fields:

> <img width="199" alt="screen shot 2018-02-17 at 6 21 51 am" src="https://user-images.githubusercontent.com/47792/36341946-ed182442-13aa-11e8-9372-a4c1d3f7746c.png">

Top level folder options:

- New subfolder

Subfolder options:

- New subfolder
- Rename folder
- Delete folder

**Comment by brandonkelly** [comment]
Ideally this should be done in Craft 2 and ported to 3.

**Comment by j-greig** [comment]
Thanks for logging this — I mentioned it in Slack today because 4 months into using Craft I've only just discovered that you can add subfolders :)
